starter-code/app.js

```diff
@@ -1,86 +1,91 @@
-var express = require('express');
-var path = require('path');
-var favicon = require('serve-favicon');
-var logger = require('morgan');
-var cookieParser = require('cookie-parser');
-var bodyParser = require('body-parser');
-var app = express();
-
-var index = require('./routes/index');
-var users = require('./routes/users');
-const passportRouter = require("./routes/passportRouter");
-//mongoose configuration
-const mongoose = require("mongoose");
-mongoose.connect("mongodb://localhost/passport-local");
-//require the user model
-const User = require("./models/user");
-const session       = require("express-session");
-const bcrypt        = require("bcrypt");
-const passport      = require("passport");
-const LocalStrategy = require("passport-local").Strategy;
-const flash = require("connect-flash");
-
-
-
-
-
-//enable sessions here
-
-
-
-
-//initialize passport and session here
-
-
-
+// require npm packages
+const express = require('express');
+const path = require('path');
+const favicon = require('serve-favicon');
+const logger = require('morgan');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+const flash = require('connect-flash');
+const passport = require('passport');
+const configurePassport = require('./helpers/passport');
+
+// run express
+const app = express();
+
+// require routes
+const index = require('./routes/index');
+const users = require('./routes/users');
+const passportRouter = require('./routes/passportRouter');
+
+// mongoose configuration
+mongoose.Promise = Promise;
+mongoose.connect('mongodb://localhost/ibi-ironhack', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
 
+// enable sessions here
+app.use(session({
+  store: new MongoStore({
+    mongooseConnection: mongoose.connection,
+    ttl: 24 * 60 * 60 // 1 day
+  }),
+  secret: 'our-passport-local-strategy-app',
+  resave: true,
+  saveUninitialized: true,
+  cookie: {
+    maxAge: 24 * 60 * 60 * 1000
+  }
+}));
```

Comment:
This session configuration should be after the cookie parser. Else it will create a new session every time you do a request.